libraries/default.rb

```diff
@@ -40,6 +40,66 @@ def self.error_message(error_code)
     end
   end
 
+  class ZookeeperConfig
+    attr_reader :config
+
+    def initialize(config: [])
+      @config = config
+    end
+
+    def self.from_h(input)
+      config = []
+      input.each do |k, v|
+        config.append({ k => v })
+      end
+      ZookeeperConfig.new(config: config)
+    end
+
+    def self.from_text(input)
+      config = input.split("\n").map do |line|
+        c = line.split('=')
+        { c[0] => c[1] }
+      end
+      ZookeeperConfig.new(config: config)
+    end
+
+    def apply!(update)
```

Comment:
I would suggest to make a copy of update to avoid side effects that are hard to debug :)